zkvm/tests/zkvm.rs

```diff
@@ -527,9 +527,8 @@ fn predicate_disjunction_happy_path() {
     let prog = Program::build(|p| {
         p.push(Output::new(prev_output))
             .input()
-            .push(key_pred)
-            .push(program_pred)
-            .select(2, 0)
+            .choose_predicate(disjunction, |t| t.select(0).map(|_| ()))
```

Comment:
I'm not sure if there's a better workaround here: effectively, we need `choose_fn` to return `Result<(), VMError>` since when we return the `Result<PredicateTree, VMError>` that comes from `select()`, there is an error in specifying lifetimes because returning the borrowed program in a new `PredicateTree` conflicts with the input lifetime

Reply:
might be worth just changing the `choose_fn` signature to specify that and adding a function other than `call` that returns the `Result<(), VMError>`, unless there's a better solution!